datadog/metric.go

```diff
@@ -94,14 +91,14 @@ func (m *Metric) StackdriverData(ctx context.Context, lastPoint time.Time, rec s
 		return nil, nil, err
 	}
 	if len(series) == 0 {
-		log.Infof(ctx, "Datadog query '%s' returned no time series", m.config.Query)
+		log.WithContext(ctx).Infof("Datadog query '%s' returned no time series", m.config.Query)
```

Comment:
logrus creates a child logger instance with WithContext(), so you _could_ do:

```go
l := log.WithContext(ctx)
```

At the beginning of your function, and then in the code:

```go
l.Debugf("message here")
```

`l` will always have the context attached.

Reply:
Ah, this is awesome. I'll solve the emulator logger issue with this as well. Thanks!